lib/ampProtection.js

```diff
@@ -0,0 +1,70 @@
+/** @module tds */
+
+const {
+    generateDNRRule
+} = require('./utils')
+
+const AMP_PROTECTION_PRIORITY = 40000
+
+/**
+ * @typedef generateAmpProtectionRulesResult
+ * @property {import('./utils').DNRRule} rule
+ * @property {object} matchDetails
+ */
+
+/**
+ * Generator to produce the declarativeNetRequest rules and corresponding match
+ * details for the given trackerAllowlist configuration.
+ * @param {object} extensionConfiguration
+ *   The extension configuration.
+ * @param {function} isRegexSupported
+ *   A function compatible with chrome.declarativeNetRequest.isRegexSupported.
+ *   See https://developer.chrome.com/docs/extensions/reference/declarativeNetRequest/#method-isRegexSupported
+ * @return {AsyncGenerator<generateAmpProtectionRulesResult>}
+ */
+async function* generateAmpProtectionRules ({ features: { ampLinks } }, isRegexSupported) {
+    if (!ampLinks ||
+        ampLinks.state !== 'enabled' ||
+        !ampLinks.settings ||
+        !ampLinks.settings.linkFormats ||
+        ampLinks.settings.linkFormats.length === 0) {
+        return
+    }
+
+    const { settings: { linkFormats: ampLinkRegexps } } = ampLinks
+    const excludedRequestDomains =
+        (ampLinks?.exceptions || []).map(({ domain }) => domain)
+
+    for (const ampLinkRegex of ampLinkRegexps) {
+        // It seems that \S (non-whitespace character class) is not always
+        // matched correctly. Luckily whitespace in URLs should be encoded
+        // anyway, so let's match . (any character) instead.
+        // TODO: Figure out minimal test case and file a Chromium bug.
+        const regexFilter = ampLinkRegex.replaceAll('\\S', '.')
```

Comment:
Yes, we could update the configuration instead, I have opened an issue for that internally on Asana. I think this is always a safe replacement, since the URL can't contain whitespace anyway - shout if you can think of something I'm missing there though!